### PR TITLE
feat: implement HTTP API client

### PR DESCRIPTION
**Pull request by Unix4ever**
The client provides methods to create schematics, Talos versions, and extensions version.

Fixes: #21

### CODE REVIEW

File: .kres.yaml [hunk 1]
```diff
@@ -12,6 +12,10 @@ spec:
       GOOS: linux
       GOARCH: arm64
 ---
+kind: golang.GoVulnCheck
```
[the remark this answers is not in the document]
Comment (Unix4ever) [reply .kres.yaml]: yeah, bad merge. Fixed

File: pkg/client/client.go [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+}
+
+// Client is the Image Factory HTTP API client.
+type Client struct {
+	address string
+	client  http.Client
+}
+
+// New creates a new Omni API client.
+func New(address string, timeout time.Duration) *Client {
```
Comment (smira) [inline pkg/client/client.go:52]: it feels it's not `address`, but rather base URL, and we could parse it here to make sure it's valid, and use `.Join` method to append patch to it?

File: pkg/client/client.go [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+	c := &Client{
+		address: address,
+		client: http.Client{
+			Timeout: timeout,
```
Comment (smira) [inline pkg/client/client.go:56]: do we actually need this `Timeout` here? should it be set on the context for each request?

We could also optionally accept `http.Client` from the outside for deep configuration options

Comment (Unix4ever) [reply pkg/client/client.go:56]: can introduce options list here, use default client if not set and something like `WithClient()` to pass it

File: pkg/client/client.go (hunk 2 above)
Comment (Unix4ever) [reply pkg/client/client.go]: I do join path in each method. Should be enough I guess

Comment (Unix4ever) [reply pkg/client/client.go]: ok, updated

**Pull request merged**